### PR TITLE
Use `dart pub` instead of `dart __deprecated pub`

### PR DESCRIPTION
**Pull request by sigurdm**
This moves away from the old pub interface.

We are no longer parsing the output of pub, so not relying on the old format. 

This takes care to run pub with `--no-analytics` to preserve behavior.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: bin/internal/shared.bat [hunk 1]
```diff
@@ -155,7 +154,7 @@ GOTO :after_subroutine
       SET /A remaining_tries=%total_tries%-1
       :retry_pub_upgrade
         ECHO Running pub upgrade... 1>&2
-        "%dart%" __deprecated_pub upgrade "%VERBOSITY%" --no-precompile
+        "%dart%" --no-analytics pub upgrade
```
Comment (christopherfujino) [inline bin/internal/shared.bat:157]: is the `--no-precompile` flag no longer needed?

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 2]
```diff
@@ -607,7 +607,7 @@ class _DefaultPub implements Pub {
         'permissions for the current user.'
       );
     }
-    return <String>[sdkPath, '__deprecated_pub', ...arguments];
+    return <String>[sdkPath, '--no-analytics', 'pub', ...arguments];
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/dart/pub.dart:610]: you don't ever want analytics from these invocations? and is `__deprecated_pub` already not sending analytics?

Comment (sigurdm) [reply packages/flutter_tools/lib/src/dart/pub.dart:610]: I think eventually we do want. But I wanted this PR to have minimal behavioral change. And we will probably have to revisit the analytics soon for the new pipeline anyway.

> is __deprecated_pub already not sending analytics?

No - that has been an outstanding issue.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/dart/pub.dart:610]: SGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

File: bin/internal/shared.bat (hunk 1 above)
Comment (sigurdm) [reply bin/internal/shared.bat:157]: No - it is the default in `dart pub get now` (for some years).

**Pull request merged**